### DISCUSSION TITLE
The argument type '$NotesTable' can't be assigned to the parameter type 'ResultSetImplementation<$NotesTable, List<Note>>'.

### DISCUSSION THREAD

**Discussion by francisnnumbi** [simolus3/drift#2753] [Q&A] [answered]
Hi,

I have 2 tables that are related : categories and notes. notes is linked to categories at categoryId.

```
class Categories extends Table{
  IntColumn get id => integer().autoIncrement()();
  TextColumn get name => text().unique()();
}
```

```
class Notes extends Table{
  IntColumn get id => integer().autoIncrement()();
  TextColumn get title => text().unique()();
 TextColumn get content => text().nullable()();
 IntColumn get categoryId => integer().references(Categories, #id).nullable()();
}
```

I have a model CategoryNotesModel class that has to retrieve a category together with its notes

```
class CategoryNotesModel {
  final Category category;
  late List<Note>? notes = <Note>[];

  CategoryNotesModel(
      {required this.category, this.notes});
}
```

Now in DAO I'm trying to join the tables based on categories table, and map to the CategoryNotesModel

```
 Stream<List<CategoryNotesModel>> watchAllCategoryNotes() => select(categories).join(
        [
          innerJoin(db.notes, db.notes.categoryId.equalsExp(categories.id)),
        ],
      ).map(
        (row) {
          return CategoryNotesModel(
            category: row.readTable(categories),
            notes: row.readTableOrNull(db.notes),
          );
        },
      ).watch();
```

But, it seems, notes being a list cannot be accepted ! 

NOTE : In the documentation, the explanation is the other way round, the scenario is that of notes table joining categories table.  That works, but it does not fulfill my requirement : that of a category with its notes !

Thanks !

**Comment by francisnnumbi** [comment, marked as answer]
I have solved the issue. I have used the package RxDart as @simolus3 suggested in another  [issue](https://github.com/simolus3/drift/issues/656#issuecomment-706364682)

 ```
Stream<List<CategoryNotesModel>> watchAllCategoryNotes() {

final categories = select(categories)
.map((category) {
          return CategoryNotesModel(category: category,);
        },).watch();

final notes = select(notes).watch();

return Rx.combineLatest2(categories, notes, (List<CategoryNotesModel> c, List<Note> n){
     return c.map((categoryNote){
           categoryNote.notes = n.where((note)=>note.categoryId == categoryNote.id).toList();
     }).toList();
   });
}
```

The package I used is [RxDart](https://pub.dev/packages/rxdart),

Please can you add this case in the documentation ?

**Reply by simolus3** [reply]
The [relationships](https://drift.simonbinder.eu/docs/examples/relationships/) example describes a similar setup already.

**Reply by francisnnumbi** [reply]
Surprisingly, yes. But how didn't I see it. I remember having read that area of documentation ! I'm ashamed of myself ! I read that, but seemed not to understand it ! Now that I go through once again, it is clear !
